src/Common/memory.h

```diff
@@ -33,65 +47,93 @@ inline ALWAYS_INLINE void * newNoExept(std::size_t size) noexcept
     return malloc(size);
 }
 
+inline ALWAYS_INLINE void * newNoExept(std::size_t size, std::align_val_t align) noexcept
+{
+    return aligned_alloc(static_cast<size_t>(align), size);
+}
+
 inline ALWAYS_INLINE void deleteImpl(void * ptr) noexcept
 {
     free(ptr);
 }
 
 #if USE_JEMALLOC
 
-inline ALWAYS_INLINE void deleteSized(void * ptr, std::size_t size) noexcept
+template <std::same_as<std::align_val_t>... TAlign>
+requires DB::OptionalArgument<TAlign...>
+inline ALWAYS_INLINE void deleteSized(void * ptr, std::size_t size, TAlign... align) noexcept
 {
     if (unlikely(ptr == nullptr))
         return;
 
-    sdallocx(ptr, size, 0);
+    if constexpr (sizeof...(TAlign) == 1)
+        sdallocx(ptr, size, MALLOCX_ALIGN(alignToSizeT(align...)));
+    else
+        sdallocx(ptr, size, 0);
 }
 
 #else
 
-inline ALWAYS_INLINE void deleteSized(void * ptr, std::size_t size [[maybe_unused]]) noexcept
+template <std::same_as<std::align_val_t>... TAlign>
+requires OptionalArgument<TAlign...>
+inline ALWAYS_INLINE void deleteSized(void * ptr, std::size_t size [[maybe_unused]], TAlign... /* align */) noexcept
 {
     free(ptr);
 }
 
 #endif
 
 #if defined(OS_LINUX)
-#   include <malloc.h>
+#    include <malloc.h>
 #elif defined(OS_DARWIN)
-#   include <malloc/malloc.h>
+#    include <malloc/malloc.h>
 #endif
 
-
-inline ALWAYS_INLINE size_t getActualAllocationSize(size_t size)
+template <std::same_as<std::align_val_t>... TAlign>
+requires DB::OptionalArgument<TAlign...>
+inline ALWAYS_INLINE size_t getActualAllocationSize(size_t size, TAlign... align)
```

Comment:
Another way to write it is `align = 0` (default arg. value), because zero alignment is not used and can be interpreted as no alignment. (Not sure if it's better).

Reply:
And another option is copy-paste :)

Reply:
Yep, wanted to avoid the runtime checks for align on each new/delete operation and copy/paste
If the code became to complex I would definitely go with the copy/paste but it turned okay in my opinion 😄